src/arenaui.cpp

```diff
@@ -176,25 +183,29 @@ void ArenaUI::sortGraphicsScene()
     pp.addRect(arenaScene->sceneRect()); // select all -- selectedItems() doesnt return group children which is a case with items()
     arenaScene->setSelectionArea(pp);
 
+    for(int k = 0; k < arenaScene->selectedItems().size(); k++)arenaScene->selectedItems()[k]->setZValue(k+1);
+
     for(int k = 0; k+1 < arenaScene->selectedItems().size(); k++)
         for(int i = k+1; i < arenaScene->selectedItems().size(); i++){
             QGraphicsItem* item1 = arenaScene->selectedItems()[k];
             QGraphicsItem* item2 = arenaScene->selectedItems()[i];
-            if(item1->boundingRect().intersects(item2->boundingRect())){
+
+            QPainterPath path1 = item1->childItems().size()? ((QCasuSceneGroup*)item1)->completeShape() : item1->shape();
+            QPainterPath path2 = item2->childItems().size()? ((QCasuSceneGroup*)item2)->completeShape() : item2->shape();
+
+            if(path1.intersects(path2)){
```

Comment:
itemGroup shape now corresponds to union if childItems shapes. Before its shape was bounding rectangle which caused ilogical selecting and intersection detection. Additionaly, when sorting sceneItems, it also takes into account line that connects grouped Items where if there is intersections between completeShape(shape+line) and Shape, sceneItem which only line intersects is painter under 2nd item (so Shape is always on top of connecting line)